command/agent/consul/service_client.go

```diff
@@ -423,6 +427,212 @@ func (o *operations) String() string {
 	return fmt.Sprintf("<%d, %d, %d, %d>", len(o.regServices), len(o.regChecks), len(o.deregServices), len(o.deregChecks))
 }
 
+type ServiceClientWrapper struct {
+	serviceClients map[string]*ServiceClient // cluster name -> client
+
+	// lock controls access to serviceClients so that we can gracefully reload
+	// Consul configuration
+	lock sync.RWMutex
+}
+
+func NewServiceClientWrapper() *ServiceClientWrapper {
+	return &ServiceClientWrapper{
+		serviceClients: map[string]*ServiceClient{},
+	}
+}
+
+func (scw *ServiceClientWrapper) AddClient(name string, client *ServiceClient) {
+	scw.lock.Lock()
+	defer scw.lock.Unlock()
+	scw.serviceClients[name] = client
+}
+
+func (scw *ServiceClientWrapper) Run() {
+	scw.lock.Lock()
+	defer scw.lock.Unlock()
+
+	for _, serviceClient := range scw.serviceClients {
+		go serviceClient.Run()
+	}
+}
+
+func (scw *ServiceClientWrapper) Shutdown() error {
+	scw.lock.Lock()
+	defer scw.lock.Unlock()
+
+	for _, serviceClient := range scw.serviceClients {
+		// TODO(tgross): we never return error from ServiceClient.Shutdown, so
+		// there's no point in returning it here either
+		_ = serviceClient.Shutdown()
+	}
+
+	return nil
+}
+
+func (scw *ServiceClientWrapper) RegisterAgent(role string, services []*structs.Service) error {
+	scw.lock.RLock()
+	defer scw.lock.RUnlock()
+
+	serviceClient, ok := scw.serviceClients[structs.ConsulDefaultCluster]
```

Comment:
Just checking my understanding, this is always `default` because that's where we want to register Nomad agents, not in any of the additional clusters?

Reply:
Right! We could conceivably change that and make it configurable in the future, but for purposes of shipping this support it seemed much easier to make Nomad always register itself with the default cluster (and less prone to accidentally split-braining the Nomad cluster across Consul clusters!).